### PR TITLE
dubbo_proxy: Add NamedProtocolConfigFactory definition

### PR DESCRIPTION
**Pull request by gengleilei**
*Description*: Add NamedProtocolConfigFactory definition
*Risk Level*: low
*Testing*:  unit test
*Docs Changes*:  N/A
*Release Notes*:  N/A
[Optional Fixes #Issue]
[Optional *Deprecated*:]

this is only part of the code for DubboProxy, the rest of the code hasn't been committed yet, complete implementation: https://github.com/gengleilei/envoy/tree/feature-dubbo-router-develop/source/extensions/filters/network/dubbo_proxy

### CODE REVIEW

File: source/extensions/filters/network/dubbo_proxy/protocol.h [hunk 1]
```diff
@@ -1,24 +1,50 @@
...
+      {ProtocolType::Dubbo, "dubbo"},
+  };
+
+  const std::string& fromType(ProtocolType type) const {
+    const auto& itor = protocolTypeNameMap.find(type);
+    if (itor != protocolTypeNameMap.end()) {
+      return itor->second;
+    }
+
+    NOT_IMPLEMENTED_GCOVR_EXCL_LINE;
```
Comment (lizan) [inline source/extensions/filters/network/dubbo_proxy/protocol.h]: nit: NOT_REACHED I think?

File: source/extensions/filters/network/dubbo_proxy/filter.cc [hunk 2]
```diff
@@ -186,7 +186,7 @@ ProtocolPtr Filter::createProtocol(ProtocolCallbacks& callback) {
   using Type = envoy::extensions::filters::network::dubbo_proxy::v2alpha1::DubboProxy;
   switch (protocol_type_) {
   case Type::Dubbo:
-    return std::make_unique<DubboProtocolImpl>(callback);
+    return NamedProtocolConfigFactory::getFactory(ProtocolType::Dubbo).createProtocol(callback);
```
Comment (lizan) [inline source/extensions/filters/network/dubbo_proxy/filter.cc]: should this be `return NamedProtocolConfigFactory::getFactory(protocol_type_).createProtocol(callback);` without switch?

File: source/extensions/filters/network/dubbo_proxy/protocol.h (hunk 1 above)
Comment (gengleilei) [reply source/extensions/filters/network/dubbo_proxy/protocol.h]: Yes, we should use NOT_REACHED.

File: source/extensions/filters/network/dubbo_proxy/filter.cc (hunk 2 above)
Comment (gengleilei) [reply source/extensions/filters/network/dubbo_proxy/filter.cc]: good, i have modified it

**Pull request merged**